webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -1420,34 +1403,50 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
       * Helper method for checking if an username is already registered.
       *
       * @param username the username of the user.
+      * @param current the current username of the user.
       * @return a [[Boolean]].
       */
-    private def userByUsernameExists(username: String): Future[Boolean] = {
-        for {
-            askString <- Future(queries.sparql.admin.txt.checkUserExistsByUsername(username = username).toString)
-            // _ = log.debug("userExists - query: {}", askString)
-
-            checkUserExistsResponse <- (storeManager ? SparqlAskRequest(askString)).mapTo[SparqlAskResponse]
-            result = checkUserExistsResponse.result
-
-        } yield result
+    private def userByUsernameExists(username: String, current: String = ""): Future[Boolean] = {
+        username match {
+            case "" => FastFuture.successful(false)
+            case `current` => FastFuture.successful(true)
+            case _ => {
+                stringFormatter.validateUsername(username, throw BadRequestException(s"The username: '${username}' contains invalid characters"))
+                for {
+                    askString <- Future(queries.sparql.admin.txt.checkUserExistsByUsername(username = username).toString)
+                    // _ = log.debug("userExists - query: {}", askString)
+
+                    checkUserExistsResponse <- (storeManager ? SparqlAskRequest(askString)).mapTo[SparqlAskResponse]
+                    result = checkUserExistsResponse.result
+
+                } yield result
+            }
+        }
     }
 
     /**
       * Helper method for checking if an email is already registered.
       *
       * @param email the email of the user.
+      * @param current the current email of the user.
       * @return a [[Boolean]].
       */
-    private def userByEmailExists(email: String): Future[Boolean] = {
-        for {
-            askString <- Future(queries.sparql.admin.txt.checkUserExistsByEmail(email = email).toString)
-            // _ = log.debug("userExists - query: {}", askString)
-
-            checkUserExistsResponse <- (storeManager ? SparqlAskRequest(askString)).mapTo[SparqlAskResponse]
-            result = checkUserExistsResponse.result
-
-        } yield result
+    private def userByEmailExists(email: String, current: String = ""): Future[Boolean] = {
```

Comment:
Please use `Option` here too instead of the empty string:

```scala
private def userByEmailExists(email: String, current: Option[String] = None)
```